### DISCUSSION TITLE
Using an Excel file as a data source.

### DISCUSSION THREAD

**Discussion by Edzzio** [eclipse-birt/birt#1394] [Q&A] [answered]
Hello, I'm a brand-new user to BIRT.

My current issue is as follows:

Whenever I add an Excel file to my report on BIRT, and export the resulting .RPTDESIGN file to a BIRT report viewer on my company's website, there are a bunch of errors, and I can't view the report. The Excel file can't be found, which is quite logical, since I don't have the Excel file on the website itself. 

What I wanted to do now is to make the Excel file in question, embedded within the report itself.

To do this, I thought of different methods :

1) Tried to create a web service using a SOAP API with the Excel file
2) Tried to create and use a JDBC server with the Excel file
3) Tried creating an Excel REST API (scrapped since BIRT only accepts SOAP based APIs)
3) Use the link to the Excel file as a data source on BIRT

Are there any other methods to resolve this issue?

I sincerely apologize if the explanation isn't very clear.

Thank you in advance.

Best regards, 
Edzzio.

**Comment by speckyspooky** [comment]
Hello Edzzio,

yes, there is a very simple way. If you create a new data source you can select the "Excel Data Source" of the JDBC-drivers - see image above.
This is a default driver of BIRT. There you can define the absolute or relative path to your file.
The next step is to reate your data set and there you can select your Excel-sheet.

The report will runs like both the excel and the sheet with the same path and same sheets exists.
Later if you need more dynamic file name or sheet you cann add additional scripts on the specific elements (Data Source: file path, Data Set: sheet name).

![excel-data-source](https://github.com/eclipse-birt/birt/assets/41593722/68e9f79b-4413-41b5-9f0f-9f6615ee3160)

**Comment by wimjongman** [comment]
Did you try deploying the Excel file in the same directory as the rptdesign file?

**Reply by Edzzio** [reply]
Hello wimjongman, 

Yes I have, the Excel file is in the same directory as the .RPTDESIGN file.

**Reply by speckyspooky** [reply]
If you have both in the same folder you can use the relativ path to the file instead of the full absolute path.
Wen your filename and the sheet name is the same you can add the path to the report and use it directly without any script.
And if you replace the excel file with the same file name & sheet name later it will works again.
The structure of your columns must be the same too.

So it should be possible to handle it on your designer & the prevview and later also in your integrated scenario.
Have you defined your report with the excel and is it runable in the designer & preview?

**Reply by Edzzio** [reply, marked as answer]
Hello yet again, 

I found the solution! What you said was correct, I chose the absolute path for the Excel file to fix one of my issues. The bigger issue was as following :

My task at hand was to create a report with different data sources and to integrate the report into the BIRT report viewer within our company's website. Our company operates on different servers, the BIRT report viewer was not operating on my local PC but on a different one. So, I had to take the Excel file in question, paste it in BIRT report viewer's folder within the server and then it was alright!

I thank you nonetheless for the help, and I wish you a wonderful day.

Best regards, 
Edzzio.

**Reply by speckyspooky** [reply]
I'm glad to hear your success!
